sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchAsyncClient.java

```diff
@@ -0,0 +1,113 @@
+import com.azure.core.implementation.util.FluxUtil;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.implementation.AzureBlobStorageBuilder;
+import com.azure.storage.blob.implementation.AzureBlobStorageImpl;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.StorageException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import static com.azure.core.implementation.util.FluxUtil.withContext;
+import static com.azure.storage.blob.implementation.PostProcessor.postProcessResponse;
+
+public final class BlobBatchAsyncClient {
+    private final AzureBlobStorageImpl client;
+
+    BlobBatchAsyncClient(String accountUrl, HttpPipeline pipeline) {
+        this.client = new AzureBlobStorageBuilder()
+            .url(accountUrl)
+            .pipeline(pipeline)
+            .build();
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If any request in a batch fails this will throw a {@link StorageException}.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.submitBatch#BlobBatch}
+     *
+     * @param batch Batch to submit.
+     * @return An empty response indicating that the batch operation has completed.
+     * @throws StorageException If any request in the {@link BlobBatch} failed or the batch request is malformed.
+     */
+    public Mono<Void> submitBatch(BlobBatch batch) {
+        return withContext(context -> submitBatchWithResponse(batch, true, context)).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Submits a batch operation.
+     *
+     * <p>If {@code throwOnAnyFailure} is {@code true} a {@link StorageException} will be thrown if any request
+     * fails.</p>
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.submitBatchWithResponse#BlobBatch-boolean}
+     *
+     * @param batch Batch to submit.
+     * @param throwOnAnyFailure Flag to indicate if an exception should be thrown if any request in the batch fails.
+     * @return A response only containing header and status code information, used to indicate that the batch operation
+     * has completed.
+     * @throws StorageException If {@code throwOnAnyFailure} is {@code true} and any request in the {@link BlobBatch}
+     * failed or the batch request is malformed.
+     */
+    public Mono<Response<Void>> submitBatchWithResponse(BlobBatch batch, boolean throwOnAnyFailure) {
+        return withContext(context -> submitBatchWithResponse(batch, throwOnAnyFailure, context));
+    }
+
+    Mono<Response<Void>> submitBatchWithResponse(BlobBatch batch, boolean throwOnAnyFailure, Context context) {
+        return postProcessResponse(this.client.services().submitBatchWithRestResponseAsync(
```

Comment:
>postProcessResponse [](start = 15, length = 19)

This, I think, is in the process of getting removed

Reply:
This will happen in preview 5 as it is a non-API breaking change and that PR can roll this up.